src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -388,15 +400,9 @@ CreateFilteredTargetListForRelation(Oid relationId, List *requiredAttributes)
 static List *
 CreateDummyTargetList(Oid relationId, List *requiredAttributes)
 {
-	Relation relation = relation_open(relationId, AccessShareLock);
-
-	Form_pg_attribute attributeTuple =
-		TupleDescAttr(relation->rd_att, 0);
-	TargetEntry *nullTargetEntry =
-		CreateTargetEntryForNullCol(attributeTuple, 1);
-
-	relation_close(relation, NoLock);
-	return list_make1(nullTargetEntry);
+	int resno = 1;
+	TargetEntry *dummyTargetEntry = CreateTargetEntryForDroppedCol(resno);
```

Comment:
Maybe CreateNullTargetEntry / CreateUnusedTargetEntry?

Reply:
Maybe I can have a wrapper around `CreateTargetEntryForDroppedCol` because `CreateTargetEntryForDroppedCol` is specifically used for dropped columns and here we should create a target entry that is similar to a target entry of a dropped column (as we should use the dummy values for the constant). So CreateUnusedTargetEntry could call `CreateTargetEntryForDroppedCol`. We already have CreateNullTargetEntry, which creates a null target entry but it has the actual column types etc.

Reply:
it's mainly weird to use CreateTargetEntryForDroppedCol for other purposes than dropped columns. Maybe the name should describe what it returns rather than what it is used for?

Reply:
Makes sense